### PR TITLE
Added visualisation Earth umbra and penumbra for satellites

### LINKED ISSUE #2360: Umbra and Penumbra based on custom distance
**Is your feature request related to a problem? Please describe.**
The Problem with the current implementation is that it seems fixed for the distance to the Moon. We operate Satellites in geostationary orbit and it would be interesting to see the umbra/penumbra based on the distance to those satellites to assess when they enter the shadow during eclipse time (March-April/September-October).

**Describe the solution you'd like**
I would like to be able to set a value for the distance the umbra-calculatuion will be based on. Alternatively, a checkbox to add the umbra lines for 36,000km distance in addition to the existing umbra indicators.

**Describe alternatives you've considered**
None possible for me

**Additional context**
I am chair of the Astronomy Club at EUMETSAT and we are very keen to observe the GEO satellites just before they enter the shadow, as they do flare up for a minute or so and become visible, whie otehrwise they never are (way too dark without direct sunlight reflection)

Would that not be an easy and interesting additional feazure? Should be quite easy to programm, as code should be same as for the other umbra circles, just with different base value for distance. Or?

We all thank you for this awesome tool!

Carsten

### PR DESCRIPTION
**Pull request by alex-w**
### Description
Added visualisation Earth umbra and penumbra for artificial satellites at fixed distance or at distance of selected satellite (distances are started counting from Earth's surface).

Fixes #2360 and #1246 (issue)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by alex-w** [comment]
@gzotti maybe you have an idea how to rearrange the GUI of Satellites plugin for making it compact?

**Comment by alex-w** [comment]
OK, I have an idea for compact GUI. Should I change it here or do it after merge?

**Comment by gzotti** [comment]
Sorry no time today. Please advance in this branch. Maybe tonight I can look into it.

**Comment by alex-w** [comment]
Damn, all umbra/penumbra calculations are topocentric!

**Comment by alex-w** [comment]
Thanks for explanation

**Comment by alex-w** [comment]
OK, I think it is ready to merge

**Comment by gzotti** [comment]
Looks interesting. However, I just followed a satellite into the shadow. It vanished far outside the shadow circle :-(

It may also be good to show one circle for the selected satellite, and another with settable distance. (Or even parse a stringlist with heights above geocenter?)

**Comment by gzotti** [comment]
To check, draw a circle in Lunar distance. Your circles are smaller. Did you apply the same shadow enlargements as for the Moon? (And obey the Danjon switch!)

**Comment by alex-w** [comment]
> Looks interesting. However, I just followed a satellite into the shadow. It vanished far outside the shadow circle :-(

I compute the shadow as geometric object - without enlargements and probably this is not very good.

> It may also be good to show one circle for the selected satellite, and another with settable distance. (Or even parse a stringlist with heights above geocenter?)

I think at the moment no need many shadow circles - I preferred to get a feedback from satellites observers, like Jay Respler

**Comment by alex-w** [comment]
> To check, draw a circle in Lunar distance. Your circles are smaller. Did you apply the same shadow enlargements as for the Moon? (And obey the Danjon switch!)

Just geometric shadow - let me change it to enlargement version

**Comment by gzotti** [comment]
You should check the code used in the Satellites plugin to determine when a satellite is in shadow. This should closely represent the shadow circle, i.e. satellite should vanish right at the circles. (Transition between penumbra and umbra, if possible!)

**Comment by alex-w** [comment]
> You should check the code used in the Satellites plugin to determine when a satellite is in shadow. This should closely represent the shadow circle, i.e. satellite should vanish right at the circles. (Transition between penumbra and umbra, if possible!)

See method `gSatWrapper::Visibility gSatWrapper::getVisibilityPredict()`

**Comment by alex-w** [comment]
> Yes, that's it - much better now! Any SUG changes?

No changes yet, because I plan to add more features

**Comment by gzotti** [comment]
It may look more natural to place the color buttons next to their actual targets, and not have 4 in a row with exclusive tooltip descriptions. (At least in earlier years, tooltips sometimes did not show!)

File: plugins/Satellites/src/gui/SatellitesDialog.cpp [hunk 1]
```diff
@@ -1048,11 +1084,17 @@ void SatellitesDialog::populateInfo()
...
 	ui->labelStdMagnitude->setToolTip(QString("<p>%1</p>").arg(q_("The standard magnitude of a satellite is defined as its apparent magnitude when at half-phase and at a distance 1000 km from the observer.")));
 	// TRANSLATORS: duration
-	ui->orbitDurationSpin->setSuffix(qc_(" s","time unit"));
-	// TRANSLATORS: duration
-	ui->labelSegmentLength->setText(q_("Segment length:"));
+	QString s = qc_("s","time unit");
+	ui->orbitDurationSpin->setSuffix(QString(" %1").arg(s));
 	// TRANSLATORS: duration
-	ui->updateFrequencySpinBox->setSuffix(qc_(" h","time unit"));
+	ui->updateFrequencySpinBox->setSuffix(QString(" %1").arg(qc_("h","time unit")));
```
Comment (gzotti) [inline plugins/Satellites/src/gui/SatellitesDialog.cpp:1090]: Try to re-use existing qc_... entries! Better write `qc_("h", "time")` and `qc_("s", "time")`.

**Review by gzotti — APPROVED** [review]
Apart from cosmetics (but please consider them) it looks OK for me.

**Pull request merged**